enumerable.rb

```diff
@@ -0,0 +1,151 @@
+module Enumerable
+  ## method to check incoming data type
+  def check_data_type(data_type)
+    arr = data_type if data_type.instance_of?(Array)
+    arr = to_a if data_type.instance_of?(Range)
+    arr = flatten if data_type.instance_of?(Hash)
+    arr
+  end
+
+  ## my_each method
+  def my_each
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    for i in arr
+      yield(i)
+    end
+  end
+
+  ## my_each_index method
+  def my_each_with_index
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    i = 0
+    until i >= arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  ## my_select method
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  ## my_any method
+  def my_any
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return true if yield(i) }
+    false
+  end
+
+  ## my_all method
+  def my_all
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return false unless yield(i) }
+    true
+  end
+
+  ## my_none method
+  def my_none
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return false unless yield(i) }
+    true
+  end
+
```

Comment:
- [ ] enumerables #my_none? returns true if the block never returns true for all elements
The method expected: true but  got: false
- [ ] enumerables #my_none? returns false if the block ever returns true for all elements
The method expected: false but  got: true
- [ ]  enumerables #my_none? when no block or argument is given returns true only if none of the collection members is true
The method expected true but  got an Array => [nil, false, nil, false]
- [ ] enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true
The method expected false but  got an Array => [nil, false, nil, false]
- [ ] enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class